content/03-reference/01-tools-and-interfaces/02-prisma-client/078-error-reference.mdx

```diff
@@ -0,0 +1,91 @@
+| `P1010`   | "User `${database_user}` was denied access on the database `${database_name}`"<br />                                                                                                                                                                                                                                                                                                                                                                                    |
+| `P1014`   | "The underlying \${kind} for model `${model}` does not exist."<br />                                                                                                                                                                                                                                                                                                                                                                                                    |
+
+#### Introspection Engine
+
+| **Error** | **Example message**                                                                      | **Description** |
+| :-------- | :--------------------------------------------------------------------------------------- | :-------------- |
+| `P4000`   | "Introspection operation failed to produce a schema file: \${introspection_error}"<br /> |
+| `P4001`   | "The introspected database was empty: \${connection_string}"<br />                       |
+| `P4002`   | "The schema of the introspected database was inconsistent: \${explanation}"<br />        |
+
+## <inlinecode>PrismaClientUnknownRequestError</inlinecode>
+
+Prisma Client throws a `PrismaClientUnknownRequestError` exception if the query engine returns an error related to a request that does not have an error code.
+
+## <inlinecode>PrismaClientRustPanicError</inlinecode>
+
+Prisma Client throws a `PrismaClientRustPanicError` exception if the underlying engine crashes and exits with a non-zero exit code. This means, that the engine crashes and exits with a non-zero exit code. In this case, the Prisma Client or the whole Node process must be restarted.
+
+## <inlinecode>PrismaClientInitializationError</inlinecode>
+
+Prisma Client throws a `PrismaClientInitializationError` exception if something goes wrong when the query engine binary is started. This happens either:
+
+- When `prisma.connect()` is called OR
+- After the first query
+
+Errors that can occur include:
+
+- The port that the query engine HTTP server wants to bind to is already taken
+- The schema fails validation because of a mising or inaccessible environment variable
```

Comment:
Is this really schema validation that is failing here?

Reply:
Hmm, no, I guess it's not really schema validation - it looks like the exception is thrown when there's an issue with the `generator` block (could not find platform). I updated the list to:

```
- The port that the query engine HTTP server wants to bind to is already taken
- A missing or inaccessible environment variable
- The query engine binary for the current platform could not be found (`generator` block)
```